### PR TITLE
Add UI support for mobile verification on update

### PR DESCRIPTION
**Pull request by DinikaSen**
## Purpose
This PR introduces mobile update wizard and mobile verification support when `mobile number verification on update` is enabled. 

demo1

Note : This feature should be disabled by default in the my-account and should be enabled only if 'mobile verification on update' feature is enabled from the management console (See [documentation](https://is.docs.wso2.com/en/latest/develop/enable-mobile-verification-for-an-updated-mobile-number/)). 

In order to enable the feature, remove `"profileInfo.mobileVerification"` from the disabled features of personal info section in `deployment.config.json`

```
    "personalInfo": {
	"disabledFeatures": ["profileInfo.mobileVerification"],
	"enabled": true,
	"scopes": {
	    "create": [],
	    "read": [],
	    "update": [],
	    "delete": []
	}
    },
```

Resolves UI part of : https://github.com/wso2/product-is/issues/9045

### CODE REVIEW

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx [hunk 1]
```diff
@@ -22,14 +22,16 @@ import { isEmpty } from "lodash";
 import React, { useEffect, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { useDispatch, useSelector } from "react-redux";
-import { Form, Grid, Icon, List } from "semantic-ui-react";
+import {Form, Grid, Icon, List, Placeholder} from "semantic-ui-react";
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx:25]:
```suggestion
import { Form, Grid, Icon, List, Placeholder } from "semantic-ui-react";
```

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx [hunk 2]
```diff
@@ -22,14 +22,16 @@ import { isEmpty } from "lodash";
...
 import { updateProfileInfo } from "../../../api";
 import { MFAIcons } from "../../../configs";
 import { CommonConstants } from "../../../constants";
-import { AlertInterface, AlertLevels, BasicProfileInterface } from "../../../models";
+import {AlertInterface, AlertLevels, BasicProfileInterface, ProfileSchema} from "../../../models";
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx:29]:
```suggestion
import { AlertInterface, AlertLevels, BasicProfileInterface, ProfileSchema } from "../../../models";
```

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx [hunk 3]
```diff
@@ -62,6 +64,10 @@ export const SMSOTPAuthenticator: React.FunctionComponent<SMSOTPProps> = (props:
         (state: any) => state.authenticationInformation.profileInfo
     );
     const activeForm: string = useSelector((state: AppState) => state.global.activeForm);
+    const [ showMobileUpdateWizard, setShowMobileUpdateWizard ] = useState<boolean>(false);
+
+    // TODO: Get from config
+    const isMobileVerificationEnabled: boolean = true;
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx:70]: Shall we resolve this TODO ?

File: apps/myaccount/src/components/profile/profile.tsx [hunk 4]
```diff
@@ -67,6 +68,10 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): J
     const [ profileSchema, setProfileSchema ] = useState<ProfileSchema[]>();
     const [ isEmailPending, setEmailPending ] = useState<boolean>(false);
     const [ showEditAvatarModal, setShowEditAvatarModal ] = useState<boolean>(false);
+    const [ showMobileUpdateWizard, setShowMobileUpdateWizard ] = useState<boolean>(false);
+
+    // TODO: Get from config
+    const isMobileVerificationEnabled: boolean = true;
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/profile/profile.tsx:74]: Let's get this from `deployment.config.json` within the app for time being.

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx (hunk 3 above)
Comment (jeradrutnam) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx:70]: Let's get this from `deployment.config.json` within the app for time being.

File: apps/myaccount/src/components/profile/profile.tsx [hunk 5]
```diff
@@ -265,7 +270,9 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): J
             const fieldName = t("userPortal:components.profile.fields." + schema.name.replace(".", "_"),
                 { defaultValue: schema.displayName }
             );
-
+            if (isMobileVerificationEnabled && checkSchemaType(schema.name, "mobile")) {
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/profile/profile.tsx:273]:
```suggestion

            if (isMobileVerificationEnabled && checkSchemaType(schema.name, "mobile")) {
```

File: apps/myaccount/src/components/profile/profile.tsx [hunk 6]
```diff
@@ -633,9 +716,26 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): J
...
+                                                    }
+                                                    wizardOpen={ true }
+                                                    currentMobileNumber={ profileInfo.get(schema.name) }
+                                                    isMobileRequired={ schema.required }
+                                                />
+                                            )
+                                            : null
+                                    }
+                                    <List.Item key={ index } className="inner-list-item">
+                                        {generateSchemaForm(schema)}
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/profile/profile.tsx:736]:
```suggestion
                                        { generateSchemaForm(schema) }
```

File: apps/myaccount/src/components/shared/mobile-update-wizard.tsx [hunk 7]
```diff
@@ -0,0 +1,421 @@
...
+
+/**
+ * Mobile number update section.
+ *
+ * @return {JSX.Element}
+ */
+export const MobileUpdateWizard: React.FunctionComponent<MobileUpdateWizardProps> =
+    (props: MobileUpdateWizardProps): JSX.Element => {
+    const { onAlertFired, closeWizard, wizardOpen, currentMobileNumber, isMobileRequired } = props;
+    const [step, setStep] = useState<number>(0);
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/shared/mobile-update-wizard.tsx:50]:
```suggestion
    const [ step, setStep ] = useState<number>(0);
```

File: apps/myaccount/src/components/shared/mobile-update-wizard.tsx [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+    const [verificationError, setVerificationError] = useState<boolean>(false);
+    const [resendSuccess, setResendSuccess] = useState<boolean>(false);
+    const { t } = useTranslation();
+    const [updateMobile, setUpdateMobile] = useTrigger();
+    const [submit, setSubmit] = useTrigger();
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/shared/mobile-update-wizard.tsx:55]:
```suggestion
    const { onAlertFired, closeWizard, wizardOpen, currentMobileNumber, isMobileRequired } = props;
    const [ step, setStep ] = useState<number>(0);
    const [ verificationError, setVerificationError ] = useState<boolean>(false);
    const [ resendSuccess, setResendSuccess ] = useState<boolean>(false);
    const { t } = useTranslation();
    const [ updateMobile, setUpdateMobile ] = useTrigger();
    const [ submit, setSubmit ] = useTrigger();
```

File: apps/myaccount/src/components/shared/mobile-update-wizard.tsx [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+            </>
+        );
+    };
+
+    /**
+     * This renders the success message at the end of the OTP flow.
+     */
+    const renderSuccess = (): JSX.Element => {
+        return (
+            <Segment basic textAlign="center">
```
Comment (jeradrutnam) [inline apps/myaccount/src/components/shared/mobile-update-wizard.tsx:348]: We already have an reusable icon for. Shall we use it here?

File: modules/i18n/src/translations/en-US/portals/user-portal.ts [hunk 10]
```diff
@@ -672,7 +672,7 @@ export const userPortal: UserPortalNS = {
             },
             smsOtp: {
                 descriptions: {
-                    hint: "You'll receive a text message containing a one-time verification code"
+                    hint: "You'll receive a text message containing a one-time verification code",
```
Comment (jeradrutnam) [inline modules/i18n/src/translations/en-US/portals/user-portal.ts:675]:
```suggestion
                    hint: "You'll receive a text message containing an one-time verification code",
```

Comment (DinikaSen) [reply modules/i18n/src/translations/en-US/portals/user-portal.ts]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/shared/mobile-update-wizard.tsx (hunk 8 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/shared/mobile-update-wizard.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/profile/profile.tsx (hunk 6 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/profile/profile.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/profile/profile.tsx (hunk 5 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/profile/profile.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/profile/profile.tsx (hunk 4 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/profile/profile.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx (hunk 3 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx (hunk 2 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx (hunk 1 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/smsotp-authenticator.tsx]: Addressed in https://github.com/wso2/identity-apps/pull/1275/commits/375a1a7a8b040739aaae46cb9b9657e461a0bf70

File: apps/myaccount/src/components/shared/mobile-update-wizard.tsx (hunk 9 above)
Comment (DinikaSen) [reply apps/myaccount/src/components/shared/mobile-update-wizard.tsx:348]: ATM we do not have a reusable icon for this.

Comment (jeradrutnam) [reply apps/myaccount/src/components/shared/mobile-update-wizard.tsx:348]: Yes, let's do this as a later improvement.

**Review by jeradrutnam — APPROVED** [review]

**Pull request merged**